File: src/main/java/sawyern/cookiebot/commands/PingCommand.java
package sawyern.cookiebot.commands;

import com.google.common.collect.Sets;
import discord4j.core.event.domain.message.MessageCreateEvent;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import sawyern.cookiebot.constants.CommandConstants;
import sawyern.cookiebot.exception.CookieException;
import sawyern.cookiebot.properties.AppProperties;

import java.util.List;
import java.util.Set;

@Component
public class PingCommand extends MessageCreateEventBotCommand {

    @Autowired
    private AppProperties appProperties;

    @Override
    public void execute(MessageCreateEvent event, List<String> args) throws CookieException {
        getBotUtil().sendMessage(event, "pong! v" + appProperties.getVersion());
    }

    @Override
    public Set<Integer> getAllowedNumArgs() {
        return Sets.newHashSet(0, 1);
    }

    @Override
    public String getCommand() {
        return CommandConstants.CommandName.PING;
    }
}

File: src/main/kotlin/sawyern/cookiebot/commands/PingCommand.kt
package sawyern.cookiebot.commands

import discord4j.core.event.domain.message.MessageCreateEvent
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import sawyern.cookiebot.constants.CommandConstants
import sawyern.cookiebot.properties.AppProperties
import sawyern.cookiebot.services.BotUtilService

@Component
class PingCommand @Autowired constructor(
        private val appProperties: AppProperties,
        botUtilService: BotUtilService
) : MessageCreateEventBotCommand(botUtilService)
{
    override fun getCommand(): String = CommandConstants.CommandName.PING
    override fun getHelpText(): String = "Checks if the server is online and returns the version"

    override fun execute(event: MessageCreateEvent, args: List<String>) {
        botUtilService.sendMessage(event, "pong! v ${appProperties.version}")
    }
}